assets/app/scripts/services/buildConfigs.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("BuildConfigsService", function(){
```

Comment:
My opinion: These methods all seem really tightly coupled to the controller. If they don't make sense outside the edit build controller, I don't think they should be put in a service.

I know you just pulled these out of the controller based on review feedback.... @benjaminapetersen and I might need to duke it out....